pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -639,8 +643,8 @@ func buildGatewayNetworkFiltersFromTLSRoutes(node *model.Proxy, env *model.Envir
 
 // Select the virtualService's hosts that match the ones specified in the gateway server's hosts
 // based on the wildcard hostname match and the namespace match
-func pickMatchingGatewayHosts(gatewayServerHosts map[model.Hostname]bool, virtualService model.Config) map[string]model.Hostname {
-	matchingHosts := make(map[string]model.Hostname, 0)
+func pickMatchingGatewayHosts(gatewayServerHosts map[model.Hostname]bool, virtualService model.Config) map[string]model.Hostnames {
+	matchingHosts := make(map[string]model.Hostnames, 0)
```

Comment:
thinking more about this, we should keep this as hostname (not the array). if a gateway has foo.com, bar.com and a virtual service has *.com with route pointing to internal.com, then what we want is two virtual hosts
foo.com
 ---> internal.com
bar.com
---> interna.com

I do agree that the code in its current form does not do that, but this change will end up being discarded immediately if we were to implement the proper logic that we discussed offline..